torch/distributed/device_mesh.py

```diff
@@ -327,8 +329,32 @@ def _setup_world_group_and_device(self):
             default_initialized = is_initialized()
             # TODO: think about how to allow pg options to be passed to world group
             # or mesh dimension groups
-            if not default_initialized:
+            use_torch_comms = int(os.environ.get("USE_TORCH_COMMS", 0))
```

Comment:
...do we really want it to be an envvar 🤔

Reply:
I don't have a better solution, torchComm itself is a separate package, originally I was thinking about using import with try-catch. But what if users want to use c10d even when torchComm is enabled? IDK..

Reply:
Is it better to require a special arg to init_device_mesh instead? Though I don't think it's hugely burdensome to require users to use a different function for init with torchcomms

I'm not a big fan of envs for this. Maybe a "lighterweight" option to the existing method in torchcomms where you just pass the backend name and dimensions rather than the full comm objects